pepdbagent/modules/group.py

```diff
@@ -0,0 +1,447 @@
+from sqlalchemy.exc import IntegrityError, NoResultFound
+from sqlalchemy.orm import Session
+from sqlalchemy import Select
+
+from peppy.const import SAMPLE_RAW_DICT_KEY, SUBSAMPLE_RAW_LIST_KEY, CONFIG_KEY
+
+from pepdbagent.const import *
+from pepdbagent.db_utils import BaseEngine, PEPGroup, GroupProjectAssociation
+from pepdbagent.exceptions import GroupUniqueNameError, GroupNotFoundError
+from pepdbagent.models import GroupListInfo, GroupInfo, GroupUpdateModel, ProjectRegistryPath
+from pepdbagent.utils import tuple_converter
+from pepdbagent.modules.project import PEPDatabaseProject
+
+
+_LOGGER = logging.getLogger(PKG_NAME)
+
+
+class PEPDatabaseGroup:
+    """
+    Class that represents Group of projects in Database.
+
+    While using this class, user can create, delete, update groups and add, remove projects from it.
+    """
+
+    def __init__(self, pep_db_engine: BaseEngine):
+        """
+        :param pep_db_engine: pepdbengine object with sa engine
+        """
+        self._sa_engine = pep_db_engine.engine
+        self._pep_db_engine = pep_db_engine
+
+    def get(
+        self,
+        namespace: str = None,
+        name: str = None,
+        query: str = None,
+        admin: Union[List[str], Tuple[str]] = tuple(),
+        limit: int = DEFAULT_LIMIT,
+        offset: int = DEFAULT_OFFSET,
+    ) -> GroupListInfo:
+        """
+        Get group of peps. (* This method doesn't retrieve actual PEPs)
+
+        There is 5 scenarios how to get Group of PEPs:
+            - provide namespace and name. Return: groups of exact provided PK(namespace, name)
+            - provide only namespace. Return: list of groups in specified namespace
+            - Nothing is provided. Return: list of groups in all database
+            - provide query. Return: list of groups find in database that have query pattern (search will be done based on name and description)
+            - provide query and namespace. Return: list of groups find in specific namespace
+                that have query pattern.
+        :param namespace: Namespace
+        :param name: Group name
+        :param query: query (search string): Pattern of name, or description
+        :param admin: admin name (namespace), or list of namespaces, where user is admin
+        :param limit: return limit
+        :param offset: return offset
+        :param order_by: sort the result-set by the information
+            Options: ["name", "update_date", "submission_date"]
+            [Default: update_date]
+        :param order_desc: Sort the records in descending order. [Default: False]
+        :return: pydantic model: GroupListInfo
+        """
+        if all([namespace, name]):
+            found_annotation = [
+                self._get_single_group(
+                    namespace=namespace,
+                    name=name,
+                    admin=admin,
+                )
+            ]
+            return GroupListInfo(
+                count=len(found_annotation),
```

Comment:
Wont this always be 1? I l like `len(found_annotation)` because it prevents hard coding, but also seems like it could cause confusion if you are looking through the code. I don't know, was just thinking.